content/6_Advanced/Comb_Sub.mdx

```diff
@@ -121,6 +121,11 @@ complexity of $\mathcal{O}(N^2)$.
 
 </Spoiler>
 
+This is another way to look at it:
+
+> Think of each subtree rooted at $i$, as a node $i$ in an undirected graph of $n$ singletons.
+> Merging two subtrees $A$ and $B$ of sizes $X$ and $Y$ is equivalent to adding at most $X\cdot Y$ edges between pairs of nodes from different subtrees.
+> After all merges, we have a complete graph with $\mathcal{O}(N^2)$ edges.
```

Comment:
maybe put this in the spoiler

Reply:
Ok, it's similar to the previous paragraph.